tensorflow/compiler/mlir/tensorflow/ir/tf_ops_a_m.cc

```diff
@@ -1246,6 +1241,139 @@ LogicalResult Conv2DOp::UpdateDataFormat(StringRef data_format) {
   return success();
 }
 
+LogicalResult Conv2DOp::inferReturnTypes(
+    mlir::MLIRContext* context, llvm::Optional<mlir::Location> location,
+    mlir::ValueRange operands, mlir::DictionaryAttr attributes,
+    mlir::RegionRange regions,
+    llvm::SmallVectorImpl<mlir::Type>& inferredReturnTypes) {
+  const int64_t num_spatial_dims = 2;
+  const int64_t num_dims = 4;
+  const Value input = operands[0];
+  const Value filter = operands[1];
+  const TensorType input_ty = input.getType().cast<TensorType>();
+  const TensorType filter_ty = filter.getType().cast<TensorType>();
+  const StringRef paddings =
+      attributes.get("padding").cast<::mlir::StringAttr>().getValue();
```

Comment:
we need to check here that attributes contains this attribute and it is of type String before cast.

Reply:
Used OpAdaptor now. Thanks